### PR TITLE
never remove id gates

### LINKED ISSUE #2373: Didn't wait time in 2-qubit circuit
### Information

- **Qiskit Terra version**: 0.8.0(master)
- **Python version**: 3.7.1
- **Operating system**: macOS Mojave 10.14.4

### What is the current behavior?
I'm measuring T1 for a specific qubit of melbourne device, qubit 0. 
If I use a single-qubit circuit it works well:
image
But if I use a 2-qubit circuit, using the same algorithm, it seems like the state didn't decay, or the identity gate aren't applied.
image

### Steps to reproduce the problem
```
# Importing needed libraries
from qiskit import *
from qiskit.mapper import Layout
import numpy as np
import matplotlib as mp
import matplotlib.pyplot as plt
from scipy.optimize import curve_fit

# Enable use of real device
IBMQ.load_accounts()
backend_exp = IBMQ.get_backend('ibmq_16_melbourne')

for u in range(0,1): # It isn't important, it is because I measured all qubit's T1
    u_st = str(u)
    file1 = 'T1__raw_qubit_' + u_st + '.txt'
    out1 = open( file1, 'w' )
    
    out1.write('# This is the qubit\'s ' + u_st +' T1 raw data \n' )
    
    circuit = []

    q = QuantumRegister(2, 'q') #Only changing this and the layout makes it works
    c1 = ClassicalRegister(1, 'c')
    qc = QuantumCircuit(q)
    mz = QuantumCircuit(q,c1)
    lay = Layout({ (q,0) : 0, (q,1):1 })

    # Exciting the qubit
    qc.x(q[0])
    qc.barrier(q)

    # Measurment on Z-axis
    mz.measure(q[0],c1[0])

    # Waiting time ( 30*0.12 us each iteration)
    for i in range(50):
        identity = QuantumCircuit(q,c1)
        identity.barrier(q)
        for k in range(i*30):
            identity.iden(q)
        identity.barrier(q)
        circuit.append(qc+identity+mz)

    # Running the experiment
    jobZ = execute(circuit, backend_exp, initial_layout=lay, shots=1024)
    
    out1.write('# N° id_gates Z-Measure Error \n')
    
    Result = jobZ.result() # Taking the results
    
    counts = []
    for i in range(50):
        counts.append(Result.get_counts(circuit[i]) )

    # Preparing the lists to make fits
    y = []
    x = []
    for i in range(50):
        py = counts[i]['1']/1024
        x.append(i*30*0.12)
        y.append( py )
        out1.write(str(i*30) + ' '+ str(py)  + '\n')       
    out1.write( '\n')
    
    def expo(x, amp, slope, high):
        y = amp*np.exp(-slope*x)+high
        return y

    x = np.array(x)
    y = np.array(y)
    err_y = np.array(err_y)
   

    params , paramscov = curve_fit(expo, x, y,p0=[1,0.02,0] )
    a =np.sqrt(np.diag(paramscov))
    out1.write('The raw T1 is ' + str(1/params[1])+ ' +- ' + str(a[1]/params[1]) + '\n')
    out1.close()
   
    
    plt.figure()
    plt.plot(x, expo(x, *params),
         label='Raw fitted function')
    plt.plot(x , y, 'ro', label= 'data')
    plt.xlabel( ' Time [us] ')
    plt.ylabel(' Probability of being in the excited state ')
    plt.legend()
    plt.savefig('plot_q_'+ u_st+ '_raw.png')
```

### What is the expected behavior?
I would expect the same behavior using 1-qubit circuit and 2-qubits circuit, because both times I put gates only on the first qubit.

### Suggested solutions
I don't know

### PR DESCRIPTION
**Pull request by ajavadia**
### Summary
Fixes #2373 

Identity gates are now like explicit 'wait' commands. They will never be removed. Even in the absence of barriers:

i.e. this circuit will stay like this even at high optimization levels, and will mean "excite qubit, then wait for 11 single-qubit-gate cycles.
image

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -225,6 +225,8 @@ The format is based on [Keep a Changelog].
 -   Added support for parameterized circuits. (\#2103)
 -   Added preset PassManagers that offer predetermined pipelines of
     transpiler passes. (\#2163)
+-   Possible to decompose SU(4) gate into non-CNOT basis with
+    `TwoQubitDecomposer`
```
Comment (mtreinish) [inline CHANGELOG.md]: This doesn't seem related to this PR, probably just a rebase or merge issue.

Comment (kdk) [inline CHANGELOG.md]: `TwoQubitBasisDecomposer`?

Comment (ajavadia) [reply CHANGELOG.md]: no it was in the wrong place (under the "fixed" heading). I moved it under the "added" heading, and also it was for an older release not current.

**Comment by ajavadia** [comment]
> Do we want to leave the definition of id in terms of U3, or make it opaque?
good point. i will make it opaque.

File: CHANGELOG.md (hunk 1 above)
Comment (mtreinish) [reply CHANGELOG.md]: This line is not about adding the `TwoQubitBasisDecomposer`, it was about a fix made to it post 0.8.0. It was added in: https://github.com/Qiskit/qiskit-terra/pull/2546 and it was about the fix being made there. (the entry is just poorly worded making this hard to figure out). The changelog entry added with the PR adding he `TwoQubitBasisDecomposer` is also not super clear, it's the first bullet here: https://github.com/Qiskit/qiskit-terra/blob/master/CHANGELOG.md#added-1 which was added in: https://github.com/Qiskit/qiskit-terra/pull/2121

But, regardless this doesn't change my point that changing this is unrelated to this PR. If we want to make independent changes to the changelog, we should do that in a separate PR to isolate this unrelated discussion from talking about the contents of this patch. Doing this separately will also makes the git log cleaner for potential revert and bisect debugging in the future.

Comment (ajavadia) [reply CHANGELOG.md]: ok i fixed it

**Comment by ajavadia** [comment]
Identity is an opaque gate now. No unrolling, no merging, no collapsing.
